### PR TITLE
Fix use live location

### PR DESCRIPTION
**Pull request by kaushiksaliya**
Update user current location.
 1. Getting user current location also getting location on background and foreground.
 2. update location every 10 seconds when current location and last location distance between 10 meters.
 
 
[Screencast from 10-07-24 04:55:47 PM IST.webm](https://github.com/canopas/yourspace-flutter/assets/140153665/7b7d46eb-8e8e-4bbb-a67f-65a17937002c) 

[Fix location tap.webm](https://github.com/canopas/yourspace-flutter/assets/140153665/26d5200a-8e4f-440d-8622-f01df946dca0)

### CODE REVIEW

File: data/lib/api/auth/api_user_service.dart [hunk 1]
```diff
@@ -186,12 +191,11 @@ class ApiUserService {
   }
 
   Future<void> signOut() async {
-    // locationManager.stopLocationTracking();
```
Comment (cp-ishita-g) [inline data/lib/api/auth/api_user_service.dart:189]: we are not tracking location?

File: data/lib/service/location_manager.dart [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import 'dart:async';
+import 'package:flutter_background_service/flutter_background_service.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:geolocator/geolocator.dart';
+import 'package:permission_handler/permission_handler.dart';
+
+import 'location_service.dart';
+
+const int locationUpdateInterval = 10000; // milliseconds
```
Comment (cp-ishita-g) [inline data/lib/service/location_manager.dart:9]: in project we are having const var in uppercase with under score, so need to make change here

File: data/lib/api/auth/api_user_service.dart (hunk 1 above)
Comment (kaushiksaliya) [reply data/lib/api/auth/api_user_service.dart:189]: if user sign out  then no need to track them.

Comment (cp-ishita-g) [reply data/lib/api/auth/api_user_service.dart:189]: i mean to say if we are tracking location than don't we need to stop tracking while sign out

Comment (kaushiksaliya) [reply data/lib/api/auth/api_user_service.dart:189]: yes, we do.

Comment (cp-ishita-g) [reply data/lib/api/auth/api_user_service.dart:189]: instead of adding you just removed code for stop location tracking, can you please add it

Comment (kaushiksaliya) [reply data/lib/api/auth/api_user_service.dart:189]: when sugnOut call then we call  locationManager.stopService(); inside this stopeService we stop live tracking and background services.

**Review by cp-ishita-g — APPROVED** [review]

**Pull request merged**